pkg/clusterapi/localcontrolplane.go

```diff
@@ -0,0 +1,158 @@
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/klog/v2"
+	capav1beta1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta1"
+	capav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	capzv1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	clusterv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3" //nolint:staticcheck
+	clusterv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4" //nolint:staticcheck
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/openshift/installer/cmd/openshift-install/command"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+func init() {
+	utilruntime.Must(clusterv1alpha3.AddToScheme(scheme.Scheme))
+	utilruntime.Must(clusterv1alpha4.AddToScheme(scheme.Scheme))
+	utilruntime.Must(clusterv1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(capav1beta1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(capav1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(capzv1.AddToScheme(scheme.Scheme))
+}
+
+// localControlPlane creates a local capi control plane
+// to use as a management cluster.
+// TODO: Add support for existing management cluster.
+type localControlPlane struct {
+	Env            *envtest.Environment
+	Client         client.Client
+	Cfg            *rest.Config
+	BinDir         string
+	KubeconfigPath string
+}
+
+// Name returns the human-friendly name of the asset.
+func (c *localControlPlane) Name() string {
+	return "Local Control Plane"
+}
+
+// Run launches the local control plane.
+func (c *localControlPlane) Run(clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	// Create a temporary directory to unpack the cluster-api binaries.
+	binDir, err := os.MkdirTemp("", "openshift-cluster-api-bins")
+	if err != nil {
+		return err
+	}
+	c.BinDir = binDir
+	if err := UnpackClusterAPIBinary(binDir); err != nil {
+		return err
+	}
+	if err := UnpackEnvtestBinaries(binDir); err != nil {
+		return err
+	}
+
+	log.SetLogger(klog.NewKlogr())
+	logrus.Info("Started local control plane with envtest")
+	c.Env = &envtest.Environment{
+		Scheme:                   scheme.Scheme,
+		AttachControlPlaneOutput: false,
+		BinaryAssetsDirectory:    c.BinDir,
+		ControlPlaneStartTimeout: 10 * time.Second,
+		ControlPlaneStopTimeout:  10 * time.Second,
+	}
+	c.Cfg, err = c.Env.Start()
+	if err != nil {
+		return err
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		// Stop the controllers when the context is cancelled.
+		<-ctx.Done()
+		if err := c.Env.Stop(); err != nil {
+			logrus.Warnf("failed to stop envtest: %v", err)
+			return
+		}
+		logrus.Info("Stopped local control plane with envtest")
+	}()
+
+	kc := fromEnvTestConfig(c.Cfg)
+	{
+		dir := filepath.Join(command.RootOpts.Dir, "auth")
```

Comment:
woah, I didn't realize `RootOpts.Dir` was a global variable. This strikes me as a little fishy but I can't think of anything really wrong with it.

Reply:
Yeah it was used in a lot of places already, and passed in `create.go` through another global (which we might remove later?)

Reply:
Sorry for the late comment, but just to track it for future development. I think that keeping `RootOpts` as a global public variable is not optimal. It's currently used within the `main` package (`cmd/` folder), and using it outside that package will increase the coupling and make harder the code to be refactored. The `localControlPlane` struct should receive as an input parameter of the `Run` command  a dir string (location of the cluster-api bin artifact), and leave to someone else the responsability to provide the right value (which should be an asset).